<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>cBioPortalData: API Reference Guide for Devs • cBioPortalData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="cBioPortalData: API Reference Guide for Devs">
<meta property="og:description" content="cBioPortalData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cBioPortalData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/cBioPortalData.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cBioPortalDataErrors.html">cBioPortalData: Data Build Errors</a>
    </li>
    <li>
      <a href="../articles/cBioPortalRClient.html">cBioPortalData: API Reference Guide for Devs</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/cBioPortalData/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="cBioPortalRClient_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>cBioPortalData: API Reference Guide for Devs</h1>
                        <h4 class="author">Marcel Ramos &amp; Levi Waldron</h4>
            
            <h4 class="date">June 18, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/cBioPortalData/blob/master/vignettes/cBioPortalRClient.Rmd"><code>vignettes/cBioPortalRClient.Rmd</code></a></small>
      <div class="hidden name"><code>cBioPortalRClient.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Please use the devel version of the <code>AnVIL</code> Bioconductor package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">cBioPortalData</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">AnVIL</span><span class="op">)</span></code></pre></div>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The cBioPortal for Cancer Genomics <a href="https://cbioportal.org">website</a> is a great resource for interactive exploration of study datasets. However, it does not easily allow the analyst to obtain and further analyze the data.</p>
<p>We’ve developed the <code>cBioPortalData</code> package to fill this need to programmatically access the data resources available on the cBioPortal.</p>
<p>The <code>cBioPortalData</code> package provides an R interface for accessing the cBioPortal study data within the Bioconductor ecosystem.</p>
<p>It downloads study data from the cBioPortal API (<a href="https://cbioportal.org/api" class="uri">https://cbioportal.org/api</a>) and uses Bioconductor infrastructure to cache and represent the data.</p>
<p>We use the <a href="https://dx.doi.org/10.1158/0008-5472.CAN-17-0344"><code>MultiAssayExperiment</code></a> (<span class="citation">@Ramos2017-er</span>) package to integrate, represent, and coordinate multiple experiments for the studies availble in the cBioPortal. This package in conjunction with <code>curatedTCGAData</code> give access to a large trove of publicly available bioinformatic data. Please see our publication <a href="https://dx.doi.org/10.1200/CCI.19.00119">here</a> (<span class="citation">@Ramos2020-ya</span>).</p>
<p>We demonstrate common use cases of <code>cBioPortalData</code> and <code>curatedTCGAData</code> during Bioconductor conference <a href="https://waldronlab.io/MultiAssayWorkshop/">workshops</a>.</p>
</div>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>This vignette is for users / developers who would like to learn more about the available data in <code>cBioPortalData</code> and to learn how to hit other endpoints in the cBioPortal API implementation. The functionality demonstrated here is used internally by the package to create integrative representations of study datasets.</p>
<p>Note. To avoid overloading the API service, the API was designed to only query a part of the study data. Therefore, the user is required to enter either a set of genes of interest or a gene panel identifier.</p>
</div>
<div id="api-representation" class="section level1">
<h1 class="hasAnchor">
<a href="#api-representation" class="anchor"></a>API representation</h1>
<p>Obtaining the cBioPortal API representation object</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">cbio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">cBioPortal</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## service: cBioPortal
## tags(); use cbioportal$&lt;tab completion&gt;:
## # A tibble: 61 x 3
##    tag           operation                  summary                             
##    &lt;chr&gt;         &lt;chr&gt;                      &lt;chr&gt;                               
##  1 Cancer Types  getAllCancerTypesUsingGET  Get all cancer types                
##  2 Cancer Types  getCancerTypeUsingGET      Get a cancer type                   
##  3 Clinical Att… fetchClinicalAttributesUs… Fetch clinical attributes           
##  4 Clinical Att… getAllClinicalAttributesI… Get all clinical attributes in the …
##  5 Clinical Att… getAllClinicalAttributesU… Get all clinical attributes         
##  6 Clinical Att… getClinicalAttributeInStu… Get specified clinical attribute    
##  7 Clinical Data fetchAllClinicalDataInStu… Fetch clinical data by patient IDs …
##  8 Clinical Data fetchClinicalDataUsingPOST Fetch clinical data by patient IDs …
##  9 Clinical Data getAllClinicalDataInStudy… Get all clinical data in a study    
## 10 Clinical Data getAllClinicalDataOfPatie… Get all clinical data of a patient …
## # … with 51 more rows
## tag values:
##   Cancer Types, Clinical Attributes, Clinical Data, Copy Number
##   Segments, Discrete Copy Number Alterations, Gene Panels, Generic
##   Assays, Genes, Molecular Data, Molecular Profiles, Mutations,
##   Patients, Sample Lists, Samples, Structural Variants, Studies,
##   Treatments
## schemas():
##   AlleleSpecificCopyNumber, AndedPatientTreatmentFilters,
##   AndedSampleTreatmentFilters, CancerStudy, CancerStudyTags
##   # ... with 56 more elements</code></pre>
<div id="operations" class="section level2">
<h2 class="hasAnchor">
<a href="#operations" class="anchor"></a>Operations</h2>
<p>Check available tags, operations, and descriptions as a <code>tibble</code>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/AnVIL/man/Services.html">tags</a></span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 61 x 3
##    tag           operation                  summary                             
##    &lt;chr&gt;         &lt;chr&gt;                      &lt;chr&gt;                               
##  1 Cancer Types  getAllCancerTypesUsingGET  Get all cancer types                
##  2 Cancer Types  getCancerTypeUsingGET      Get a cancer type                   
##  3 Clinical Att… fetchClinicalAttributesUs… Fetch clinical attributes           
##  4 Clinical Att… getAllClinicalAttributesI… Get all clinical attributes in the …
##  5 Clinical Att… getAllClinicalAttributesU… Get all clinical attributes         
##  6 Clinical Att… getClinicalAttributeInStu… Get specified clinical attribute    
##  7 Clinical Data fetchAllClinicalDataInStu… Fetch clinical data by patient IDs …
##  8 Clinical Data fetchClinicalDataUsingPOST Fetch clinical data by patient IDs …
##  9 Clinical Data getAllClinicalDataInStudy… Get all clinical data in a study    
## 10 Clinical Data getAllClinicalDataOfPatie… Get all clinical data of a patient …
## # … with 51 more rows</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/AnVIL/man/Services.html">tags</a></span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span><span class="op">$</span><span class="va">operation</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "getAllCancerTypesUsingGET"              
## [2] "getCancerTypeUsingGET"                  
## [3] "fetchClinicalAttributesUsingPOST"       
## [4] "getAllClinicalAttributesInStudyUsingGET"
## [5] "getAllClinicalAttributesUsingGET"       
## [6] "getClinicalAttributeInStudyUsingGET"</code></pre>
<div id="searching-through-the-api" class="section level3">
<h3 class="hasAnchor">
<a href="#searching-through-the-api" class="anchor"></a>Searching through the API</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">searchOps</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"clinical"</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "getAllClinicalAttributesUsingGET"          
## [2] "fetchClinicalAttributesUsingPOST"          
## [3] "fetchClinicalDataUsingPOST"                
## [4] "getAllClinicalAttributesInStudyUsingGET"   
## [5] "getClinicalAttributeInStudyUsingGET"       
## [6] "getAllClinicalDataInStudyUsingGET"         
## [7] "fetchAllClinicalDataInStudyUsingPOST"      
## [8] "getAllClinicalDataOfPatientInStudyUsingGET"
## [9] "getAllClinicalDataOfSampleInStudyUsingGET"</code></pre>
</div>
</div>
<div id="studies" class="section level2">
<h2 class="hasAnchor">
<a href="#studies" class="anchor"></a>Studies</h2>
<p>Get the list of studies available:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">getStudies</a></span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 313 x 13
##    name     shortName  description    publicStudy pmid    citation groups status
##    &lt;chr&gt;    &lt;chr&gt;      &lt;chr&gt;          &lt;lgl&gt;       &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;   &lt;int&gt;
##  1 Pan-Lun… NSCLC (TC… "Whole-exome … TRUE        271587… TCGA, N… ""          0
##  2 Gliobla… GBM (TCGA) "TCGA Gliobla… TRUE        &lt;NA&gt;    &lt;NA&gt;     "PUBL…      0
##  3 Head an… Head &amp; ne… "TCGA Head an… TRUE        &lt;NA&gt;    &lt;NA&gt;     "PUBL…      0
##  4 Breast … Breast (T… "Whole-exome … TRUE        264514… TCGA, C… "PUBL…      0
##  5 Ovarian… Ovarian (… "Whole exome … TRUE        217203… TCGA, N… "PUBL…      0
##  6 Uterine… Uterine (… "Whole exome … TRUE        236363… TCGA, N… "PUBL…      0
##  7 Bladder… BLCA (TCG… "Whole-exome … TRUE        289887… Roberts… "PUBL…      0
##  8 Head an… Head &amp; ne… "Head and Nec… TRUE        296250… TCGA, C… "PUBL…      0
##  9 Lung Ad… Lung aden… "Lung Adenoca… TRUE        296250… TCGA, C… "PUBL…      0
## 10 Lung Sq… Lung squ … "Lung Squamou… TRUE        296250… TCGA, C… "PUBL…      0
## # … with 303 more rows, and 5 more variables: importDate &lt;chr&gt;,
## #   allSampleCount &lt;int&gt;, studyId &lt;chr&gt;, cancerTypeId &lt;chr&gt;,
## #   referenceGenome &lt;chr&gt;</code></pre>
</div>
<div id="clinical-data" class="section level2">
<h2 class="hasAnchor">
<a href="#clinical-data" class="anchor"></a>Clinical Data</h2>
<p>Obtain the clinical data for a particular study:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">clinicalData</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga"</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 92 x 84
##    patientId  AGE   AJCC_PATHOLOGIC_TU… ATYPICAL_MITOTIC_FI… CAPSULAR_INVASION  
##    &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;               &lt;chr&gt;                &lt;chr&gt;              
##  1 TCGA-OR-A… 58    Stage II            Atypical Mitotic Fi… Invasion of Tumor …
##  2 TCGA-OR-A… 44    Stage IV            Atypical Mitotic Fi… Invasion of Tumor …
##  3 TCGA-OR-A… 23    Stage III           Atypical Mitotic Fi… Invasion of Tumor …
##  4 TCGA-OR-A… 23    Stage IV            Atypical Mitotic Fi… Invasion of Tumor …
##  5 TCGA-OR-A… 30    Stage III           Atypical Mitotic Fi… Invasion of Tumor …
##  6 TCGA-OR-A… 29    Stage II            Atypical Mitotic Fi… Invasion of Tumor …
##  7 TCGA-OR-A… 30    Stage III           Atypical Mitotic Fi… Invasion of Tumor …
##  8 TCGA-OR-A… 66    Stage III           Atypical Mitotic Fi… Invasion of Tumor …
##  9 TCGA-OR-A… 22    Stage II            Atypical Mitotic Fi… Invasion of Tumor …
## 10 TCGA-OR-A… 53    Stage IV            Atypical Mitotic Fi… Invasion of Tumor …
## # … with 82 more rows, and 79 more variables: CLIN_M_STAGE &lt;chr&gt;,
## #   CT_SCAN_PREOP_RESULTS &lt;chr&gt;,
## #   CYTOPLASM_PRESENCE_LESS_THAN_EQUAL_25_PERCENT &lt;chr&gt;,
## #   DAYS_TO_INITIAL_PATHOLOGIC_DIAGNOSIS &lt;chr&gt;, DFS_MONTHS &lt;chr&gt;,
## #   DFS_STATUS &lt;chr&gt;, DIFFUSE_ARCHITECTURE &lt;chr&gt;, ETHNICITY &lt;chr&gt;,
## #   FORM_COMPLETION_DATE &lt;chr&gt;, HISTOLOGICAL_DIAGNOSIS &lt;chr&gt;,
## #   HISTORY_ADRENAL_HORMONE_EXCESS &lt;chr&gt;,
## #   HISTORY_BASIS_ADRENAL_HORMONE_DX &lt;chr&gt;, HISTORY_NEOADJUVANT_TRTYN &lt;chr&gt;,
## #   HISTORY_OTHER_MALIGNANCY &lt;chr&gt;, ICD_10 &lt;chr&gt;, ICD_O_3_HISTOLOGY &lt;chr&gt;,
## #   ICD_O_3_SITE &lt;chr&gt;, INFORMED_CONSENT_VERIFIED &lt;chr&gt;,
## #   INITIAL_PATHOLOGIC_DX_YEAR &lt;chr&gt;, LATERALITY &lt;chr&gt;,
## #   LYMPH_NODES_EXAMINED &lt;chr&gt;, MITOSES_PER_50_HPF &lt;chr&gt;, MITOTIC_RATE &lt;chr&gt;,
## #   MOLECULAR_STUDIES_OTHERS_PERFORMED &lt;chr&gt;, NECROSIS &lt;chr&gt;,
## #   NEW_TUMOR_EVENT_AFTER_INITIAL_TREATMENT &lt;chr&gt;, NUCLEAR_GRADE_III_IV &lt;chr&gt;,
## #   OS_MONTHS &lt;chr&gt;, OS_STATUS &lt;chr&gt;, OTHER_PATIENT_ID &lt;chr&gt;,
## #   PATH_N_STAGE &lt;chr&gt;, PATH_T_STAGE &lt;chr&gt;, PHARMACEUTICAL_TX_ADJUVANT &lt;chr&gt;,
## #   PHARM_TX_MITOTANE_INDICATOR &lt;chr&gt;, PROSPECTIVE_COLLECTION &lt;chr&gt;,
## #   RACE &lt;chr&gt;, RADIATION_TREATMENT_ADJUVANT &lt;chr&gt;, RESIDUAL_TUMOR &lt;chr&gt;,
## #   RETROSPECTIVE_COLLECTION &lt;chr&gt;, SAMPLE_COUNT &lt;chr&gt;, SEX &lt;chr&gt;,
## #   SINUSOID_INVASION &lt;chr&gt;, SITE_OF_TUMOR_TISSUE &lt;chr&gt;,
## #   TISSUE_SOURCE_SITE &lt;chr&gt;, TUMOR_STATUS &lt;chr&gt;, WEISS_SCORE_OVERALL &lt;chr&gt;,
## #   WEISS_VENOUS_INVASION &lt;chr&gt;, CLINICAL_STATUS_WITHIN_3_MTHS_SURGERY &lt;chr&gt;,
## #   LYMPH_NODES_EXAMINED_HE_COUNT &lt;chr&gt;, LYMPH_NODE_EXAMINED_COUNT &lt;chr&gt;,
## #   METASTATIC_DX_CONFIRMED_BY &lt;chr&gt;, METASTATIC_SITE_PATIENT &lt;chr&gt;,
## #   PHARM_TX_MITOTANE_FOR_MACRO_DISEASE &lt;chr&gt;,
## #   TREATMENT_OUTCOME_FIRST_COURSE &lt;chr&gt;, DAYS_LAST_FOLLOWUP &lt;chr&gt;,
## #   PHARM_TX_MITOTANE_ADJUVANT &lt;chr&gt;,
## #   PHARM_TX_MITOTANE_THERAPUTIC_AT_REC &lt;chr&gt;,
## #   PHARM_TX_MITOTANE_THERAPUTIC_LEVELS &lt;chr&gt;,
## #   METASTATIC_DX_CONFIRMED_BY_OTHER &lt;chr&gt;,
## #   PHARM_TX_MITOTANE_THERAPUTIC_MACRO &lt;chr&gt;,
## #   PHARM_TX_MITOTANE_THERAPUTIC_AT_PROG &lt;chr&gt;, RET &lt;chr&gt;, sampleId &lt;chr&gt;,
## #   CANCER_TYPE &lt;chr&gt;, CANCER_TYPE_DETAILED &lt;chr&gt;, DAYS_TO_COLLECTION &lt;chr&gt;,
## #   FRACTION_GENOME_ALTERED &lt;chr&gt;, IS_FFPE &lt;chr&gt;, MUTATION_COUNT &lt;chr&gt;,
## #   OCT_EMBEDDED &lt;chr&gt;, ONCOTREE_CODE &lt;chr&gt;, OTHER_SAMPLE_ID &lt;chr&gt;,
## #   PATHOLOGY_REPORT_FILE_NAME &lt;chr&gt;, PATHOLOGY_REPORT_UUID &lt;chr&gt;,
## #   SAMPLE_INITIAL_WEIGHT &lt;chr&gt;, SAMPLE_TYPE &lt;chr&gt;, SAMPLE_TYPE_ID &lt;chr&gt;,
## #   SOMATIC_STATUS &lt;chr&gt;, VIAL_NUMBER &lt;chr&gt;</code></pre>
</div>
<div id="molecular-profiles" class="section level2">
<h2 class="hasAnchor">
<a href="#molecular-profiles" class="anchor"></a>Molecular Profiles</h2>
<p>A table of molecular profiles for a particular study can be obtained by running the following:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mols</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">molecularProfiles</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga"</span><span class="op">)</span>
<span class="va">mols</span><span class="op">[[</span><span class="st">"molecularProfileId"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<pre><code>## [1] "acc_tcga_rppa"                                     
## [2] "acc_tcga_rppa_Zscores"                             
## [3] "acc_tcga_gistic"                                   
## [4] "acc_tcga_rna_seq_v2_mrna"                          
## [5] "acc_tcga_rna_seq_v2_mrna_median_Zscores"           
## [6] "acc_tcga_linear_CNA"                               
## [7] "acc_tcga_methylation_hm450"                        
## [8] "acc_tcga_mutations"                                
## [9] "acc_tcga_rna_seq_v2_mrna_median_all_sample_Zscores"</code></pre>
</div>
<div id="molecular-profile-data" class="section level2">
<h2 class="hasAnchor">
<a href="#molecular-profile-data" class="anchor"></a>Molecular Profile Data</h2>
<p>The data for a molecular profile can be obtained with prior knowledge of available <code>entrezGeneIds</code>:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">molecularData</a></span><span class="op">(</span><span class="va">cbio</span>, molecularProfileId <span class="op">=</span> <span class="st">"acc_tcga_rna_seq_v2_mrna"</span>,
    entrezGeneIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,
    sampleIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TCGA-OR-A5J1-01"</span>,  <span class="st">"TCGA-OR-A5J2-01"</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<pre><code>## $acc_tcga_rna_seq_v2_mrna
## # A tibble: 4 x 8
##   uniqueSampleKey    uniquePatientKey   entrezGeneId molecularProfile… sampleId 
##   &lt;chr&gt;              &lt;chr&gt;                     &lt;int&gt; &lt;chr&gt;             &lt;chr&gt;    
## 1 VENHQS1PUi1BNUoxL… VENHQS1PUi1BNUoxO…            1 acc_tcga_rna_seq… TCGA-OR-…
## 2 VENHQS1PUi1BNUoxL… VENHQS1PUi1BNUoxO…            2 acc_tcga_rna_seq… TCGA-OR-…
## 3 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO…            1 acc_tcga_rna_seq… TCGA-OR-…
## 4 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO…            2 acc_tcga_rna_seq… TCGA-OR-…
## # … with 3 more variables: patientId &lt;chr&gt;, studyId &lt;chr&gt;, value &lt;dbl&gt;</code></pre>
</div>
<div id="genes" class="section level2">
<h2 class="hasAnchor">
<a href="#genes" class="anchor"></a>Genes</h2>
<div id="all-available-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#all-available-genes" class="anchor"></a>All available genes</h3>
<p>A list of all the genes provided by the API service including hugo symbols, and entrez gene IDs can be obtained by using the <code>geneTable</code> function:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">geneTable</a></span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 1,000 x 3
##    entrezGeneId hugoGeneSymbol type          
##           &lt;int&gt; &lt;chr&gt;          &lt;chr&gt;         
##  1       -93764 IVNS1ABP_PT330 phosphoprotein
##  2       -93763 IVNS1ABP_PT328 phosphoprotein
##  3       -93762 IVNS1ABP_PS329 phosphoprotein
##  4       -93761 IVNS1ABP_PS277 phosphoprotein
##  5       -93760 MORC2_PS785    phosphoprotein
##  6       -93759 MORC2_PS779    phosphoprotein
##  7       -93758 MORC2_PS777    phosphoprotein
##  8       -93757 MORC2_PS743    phosphoprotein
##  9       -93756 MORC2_PS739    phosphoprotein
## 10       -93755 MORC2_PS725    phosphoprotein
## # … with 990 more rows</code></pre>
</div>
<div id="gene-panels" class="section level3">
<h3 class="hasAnchor">
<a href="#gene-panels" class="anchor"></a>Gene Panels</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">genePanels</a></span><span class="op">(</span><span class="va">cbio</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 52 x 2
##    description                                               genePanelId        
##    &lt;chr&gt;                                                     &lt;chr&gt;              
##  1 Targeted (27 cancer genes) sequencing of adenoid cystic … ACYC_FMI_27        
##  2 Targeted panel of 8 genes.                                AmpliSeq           
##  3 ARCHER-SOLID Gene Panel (62 genes)                        ARCHER-SOLID-CV1   
##  4 Targeted sequencing of various tumor types via bait v3.   bait_v3            
##  5 Targeted sequencing of various tumor types via bait v4.   bait_v4            
##  6 Targeted sequencing of various tumor types via bait v5.   bait_v5            
##  7 Foundation Medicine T5a gene panel (323 genes)            FMI-T5a            
##  8 Foundation Medicine T7 gene panel (429 genes)             FMI-T7             
##  9 Foundation Medicine T5 gene panel (326 genes)             glioma_mskcc_2019_…
## 10 Foundation Medicine T7 gene panel (434 genes)             glioma_mskcc_2019_…
## # … with 42 more rows</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">getGenePanel</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"IMPACT341"</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 341 x 2
##    entrezGeneId hugoGeneSymbol
##           &lt;int&gt; &lt;chr&gt;         
##  1           25 ABL1          
##  2        84142 ABRAXAS1      
##  3          207 AKT1          
##  4          208 AKT2          
##  5        10000 AKT3          
##  6          238 ALK           
##  7          242 ALOX12B       
##  8       139285 AMER1         
##  9          324 APC           
## 10          367 AR            
## # … with 331 more rows</code></pre>
</div>
</div>
<div id="molecular-gene-panels" class="section level2">
<h2 class="hasAnchor">
<a href="#molecular-gene-panels" class="anchor"></a>Molecular Gene Panels</h2>
<div id="genepanelmolecular" class="section level3">
<h3 class="hasAnchor">
<a href="#genepanelmolecular" class="anchor"></a>genePanelMolecular</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gprppa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">genePanelMolecular</a></span><span class="op">(</span><span class="va">cbio</span>,
    molecularProfileId <span class="op">=</span> <span class="st">"acc_tcga_rppa"</span>,
    sampleListId <span class="op">=</span> <span class="st">"acc_tcga_all"</span><span class="op">)</span>
<span class="va">gprppa</span></code></pre></div>
<pre><code>## # A tibble: 92 x 7
##    uniqueSampleKey  uniquePatientKey molecularProfil… sampleId patientId studyId
##    &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;    &lt;chr&gt;     &lt;chr&gt;  
##  1 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_rppa    TCGA-OR… TCGA-OR-… acc_tc…
##  2 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_rppa    TCGA-OR… TCGA-OR-… acc_tc…
##  3 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_rppa    TCGA-OR… TCGA-OR-… acc_tc…
##  4 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_rppa    TCGA-OR… TCGA-OR-… acc_tc…
##  5 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_rppa    TCGA-OR… TCGA-OR-… acc_tc…
##  6 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_rppa    TCGA-OR… TCGA-OR-… acc_tc…
##  7 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_rppa    TCGA-OR… TCGA-OR-… acc_tc…
##  8 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_rppa    TCGA-OR… TCGA-OR-… acc_tc…
##  9 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_rppa    TCGA-OR… TCGA-OR-… acc_tc…
## 10 VENHQS1PUi1BNUp… VENHQS1PUi1BNUp… acc_tcga_rppa    TCGA-OR… TCGA-OR-… acc_tc…
## # … with 82 more rows, and 1 more variable: profiled &lt;lgl&gt;</code></pre>
</div>
<div id="getgenepanelmolecular" class="section level3">
<h3 class="hasAnchor">
<a href="#getgenepanelmolecular" class="anchor"></a>getGenePanelMolecular</h3>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">getGenePanelMolecular</a></span><span class="op">(</span><span class="va">cbio</span>,
    molecularProfileIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"acc_tcga_rppa"</span>, <span class="st">"acc_tcga_gistic"</span><span class="op">)</span>,
    sampleIds <span class="op">=</span> <span class="fu"><a href="../reference/cBioPortal.html">allSamples</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga"</span><span class="op">)</span><span class="op">$</span><span class="va">sampleId</span>
<span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 184 x 7
##    uniqueSampleKey  uniquePatientKey molecularProfil… sampleId patientId studyId
##    &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;    &lt;chr&gt;     &lt;chr&gt;  
##  1 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_gistic  TCGA-OR… TCGA-OR-… acc_tc…
##  2 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_gistic  TCGA-OR… TCGA-OR-… acc_tc…
##  3 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_gistic  TCGA-OR… TCGA-OR-… acc_tc…
##  4 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_gistic  TCGA-OR… TCGA-OR-… acc_tc…
##  5 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_gistic  TCGA-OR… TCGA-OR-… acc_tc…
##  6 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_gistic  TCGA-OR… TCGA-OR-… acc_tc…
##  7 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_gistic  TCGA-OR… TCGA-OR-… acc_tc…
##  8 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_gistic  TCGA-OR… TCGA-OR-… acc_tc…
##  9 VENHQS1PUi1BNUo… VENHQS1PUi1BNUo… acc_tcga_gistic  TCGA-OR… TCGA-OR-… acc_tc…
## 10 VENHQS1PUi1BNUp… VENHQS1PUi1BNUp… acc_tcga_gistic  TCGA-OR… TCGA-OR-… acc_tc…
## # … with 174 more rows, and 1 more variable: profiled &lt;lgl&gt;</code></pre>
</div>
<div id="getdatabygenepanel" class="section level3">
<h3 class="hasAnchor">
<a href="#getdatabygenepanel" class="anchor"></a>getDataByGenePanel</h3>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">getDataByGenePanel</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga"</span>, genePanelId <span class="op">=</span> <span class="st">"IMPACT341"</span>,
    molecularProfileId <span class="op">=</span> <span class="st">"acc_tcga_rppa"</span>, sampleListId <span class="op">=</span> <span class="st">"acc_tcga_rppa"</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: 'getDataByGenePanel' is deprecated.
## Use 'getDataByGenes' instead.
## See help("Deprecated")</code></pre>
<pre><code>## $acc_tcga_rppa
## # A tibble: 2,622 x 9
##    uniqueSampleKey    uniquePatientKey   entrezGeneId molecularProfil… sampleId 
##    &lt;chr&gt;              &lt;chr&gt;                     &lt;int&gt; &lt;chr&gt;            &lt;chr&gt;    
##  1 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO…         5728 acc_tcga_rppa    TCGA-OR-…
##  2 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO…          595 acc_tcga_rppa    TCGA-OR-…
##  3 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO…          596 acc_tcga_rppa    TCGA-OR-…
##  4 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO…        10413 acc_tcga_rppa    TCGA-OR-…
##  5 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO…         3791 acc_tcga_rppa    TCGA-OR-…
##  6 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO…         7157 acc_tcga_rppa    TCGA-OR-…
##  7 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO…          207 acc_tcga_rppa    TCGA-OR-…
##  8 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO…          208 acc_tcga_rppa    TCGA-OR-…
##  9 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO…        57521 acc_tcga_rppa    TCGA-OR-…
## 10 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO…         2064 acc_tcga_rppa    TCGA-OR-…
## # … with 2,612 more rows, and 4 more variables: patientId &lt;chr&gt;, studyId &lt;chr&gt;,
## #   value &lt;dbl&gt;, hugoGeneSymbol &lt;chr&gt;</code></pre>
<p>It uses the <code>getAllGenesUsingGET</code> function from the API.</p>
</div>
</div>
<div id="samples" class="section level2">
<h2 class="hasAnchor">
<a href="#samples" class="anchor"></a>Samples</h2>
<div id="sample-list-identifiers" class="section level3">
<h3 class="hasAnchor">
<a href="#sample-list-identifiers" class="anchor"></a>Sample List Identifiers</h3>
<p>To display all available sample list identifiers for a particular study ID, one can use the <code>sampleLists</code> function:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">sampleLists</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga"</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 9 x 5
##   category          name          description            sampleListId    studyId
##   &lt;chr&gt;             &lt;chr&gt;         &lt;chr&gt;                  &lt;chr&gt;           &lt;chr&gt;  
## 1 all_cases_with_m… Samples with… Samples with mutation… acc_tcga_seque… acc_tc…
## 2 all_cases_with_r… Samples with… Samples protein data … acc_tcga_rppa   acc_tc…
## 3 all_cases_with_m… Samples with… Samples with methylat… acc_tcga_methy… acc_tc…
## 4 all_cases_with_m… Samples with… Samples with mutation… acc_tcga_cnaseq acc_tc…
## 5 all_cases_with_m… Samples with… Samples with methylat… acc_tcga_methy… acc_tc…
## 6 all_cases_in_stu… All samples   All samples (92 sampl… acc_tcga_all    acc_tc…
## 7 all_cases_with_c… Samples with… Samples with CNA data… acc_tcga_cna    acc_tc…
## 8 all_cases_with_m… Samples with… Samples with mRNA exp… acc_tcga_rna_s… acc_tc…
## 9 all_cases_with_m… Complete sam… Samples with mutation… acc_tcga_3way_… acc_tc…</code></pre>
</div>
<div id="sample-identifiers" class="section level3">
<h3 class="hasAnchor">
<a href="#sample-identifiers" class="anchor"></a>Sample Identifiers</h3>
<p>One can obtain the barcodes / identifiers for each sample using a specific sample list identifier, in this case we want all the copy number alteration samples:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">samplesInSampleLists</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga_cna"</span><span class="op">)</span></code></pre></div>
<pre><code>## CharacterList of length 1
## [["acc_tcga_cna"]] TCGA-OR-A5J1-01 TCGA-OR-A5J2-01 ... TCGA-PK-A5HC-01</code></pre>
<p>This returns a <code>CharacterList</code> of all identifiers for each sample list identifier input:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">samplesInSampleLists</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"acc_tcga_cna"</span>, <span class="st">"acc_tcga_cnaseq"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## CharacterList of length 2
## [["acc_tcga_cna"]] TCGA-OR-A5J1-01 TCGA-OR-A5J2-01 ... TCGA-PK-A5HC-01
## [["acc_tcga_cnaseq"]] TCGA-OR-A5J1-01 TCGA-OR-A5J2-01 ... TCGA-PK-A5HC-01</code></pre>
</div>
<div id="all-samples-within-a-study-id" class="section level3">
<h3 class="hasAnchor">
<a href="#all-samples-within-a-study-id" class="anchor"></a>All samples within a study ID</h3>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">allSamples</a></span><span class="op">(</span><span class="va">cbio</span>, <span class="st">"acc_tcga"</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 92 x 6
##    uniqueSampleKey    uniquePatientKey   sampleType  sampleId  patientId studyId
##    &lt;chr&gt;              &lt;chr&gt;              &lt;chr&gt;       &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;  
##  1 VENHQS1PUi1BNUoxL… VENHQS1PUi1BNUoxO… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  2 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  3 VENHQS1PUi1BNUozL… VENHQS1PUi1BNUozO… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  4 VENHQS1PUi1BNUo0L… VENHQS1PUi1BNUo0O… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  5 VENHQS1PUi1BNUo1L… VENHQS1PUi1BNUo1O… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  6 VENHQS1PUi1BNUo2L… VENHQS1PUi1BNUo2O… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  7 VENHQS1PUi1BNUo3L… VENHQS1PUi1BNUo3O… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  8 VENHQS1PUi1BNUo4L… VENHQS1PUi1BNUo4O… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  9 VENHQS1PUi1BNUo5L… VENHQS1PUi1BNUo5O… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
## 10 VENHQS1PUi1BNUpBL… VENHQS1PUi1BNUpBO… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
## # … with 82 more rows</code></pre>
</div>
<div id="info-on-samples" class="section level3">
<h3 class="hasAnchor">
<a href="#info-on-samples" class="anchor"></a>Info on Samples</h3>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/cBioPortal.html">getSampleInfo</a></span><span class="op">(</span><span class="va">cbio</span>, studyId <span class="op">=</span> <span class="st">"acc_tcga"</span>,
    sampleListIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"acc_tcga_rppa"</span>, <span class="st">"acc_tcga_gistic"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 46 x 6
##    uniqueSampleKey    uniquePatientKey   sampleType  sampleId  patientId studyId
##    &lt;chr&gt;              &lt;chr&gt;              &lt;chr&gt;       &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;  
##  1 VENHQS1PUi1BNUoyL… VENHQS1PUi1BNUoyO… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  2 VENHQS1PUi1BNUozL… VENHQS1PUi1BNUozO… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  3 VENHQS1PUi1BNUo2L… VENHQS1PUi1BNUo2O… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  4 VENHQS1PUi1BNUo3L… VENHQS1PUi1BNUo3O… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  5 VENHQS1PUi1BNUo4L… VENHQS1PUi1BNUo4O… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  6 VENHQS1PUi1BNUo5L… VENHQS1PUi1BNUo5O… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  7 VENHQS1PUi1BNUpBL… VENHQS1PUi1BNUpBO… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  8 VENHQS1PUi1BNUpQL… VENHQS1PUi1BNUpQO… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
##  9 VENHQS1PUi1BNUpSL… VENHQS1PUi1BNUpSO… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
## 10 VENHQS1PUi1BNUpTL… VENHQS1PUi1BNUpTO… Primary So… TCGA-OR-… TCGA-OR-… acc_tc…
## # … with 36 more rows</code></pre>
</div>
</div>
</div>
<div id="advanced-usage" class="section level1">
<h1 class="hasAnchor">
<a href="#advanced-usage" class="anchor"></a>Advanced Usage</h1>
<p>The <code>cBioPortal</code> API representation is not limited to the functions provided in the package. Users who wish to make use of any of the endpoints provided by the API specification should use the dollar sign <code><a href="https://rdrr.io/r/base/Extract.html">$</a></code> function to access the endpoints.</p>
<p>First the user should see the input for a particular endpoint as detailed in the API:</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cbio</span><span class="op">$</span><span class="va">getGeneUsingGET</span></code></pre></div>
<pre><code>## getGeneUsingGET 
## Get a gene 
## 
## Parameters:
##   geneId (string)
##     Entrez Gene ID or Hugo Gene Symbol e.g. 1 or A1BG</code></pre>
<p>Then the user can provide such input:</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">resp</span> <span class="op">&lt;-</span> <span class="va">cbio</span><span class="op">$</span><span class="fu">getGeneUsingGET</span><span class="op">(</span><span class="st">"BRCA1"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Response [https://www.cbioportal.org/api/genes/BRCA1]
##   Date: 2021-06-18 20:38
##   Status: 200
##   Content-Type: application/json
##   Size: 69 B</code></pre>
<p>which will require the user to ‘translate’ the response using <code><a href="https://httr.r-lib.org/reference/content.html">httr::content</a></code>:</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">httr</span><span class="fu">::</span><span class="fu"><a href="https://httr.r-lib.org/reference/content.html">content</a></span><span class="op">(</span><span class="va">resp</span><span class="op">)</span></code></pre></div>
<pre><code>## $entrezGeneId
## [1] 672
## 
## $hugoGeneSymbol
## [1] "BRCA1"
## 
## $type
## [1] "protein-coding"</code></pre>
<div id="clearing-the-cache" class="section level2">
<h2 class="hasAnchor">
<a href="#clearing-the-cache" class="anchor"></a>Clearing the cache</h2>
<p>For users who wish to clear the entire <code>cBioPortalData</code> cache, it is recommended that they use:</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/unlink.html">unlink</a></span><span class="op">(</span><span class="st">"~/.cache/cBioPortalData/"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="sessioninfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>sessionInfo</h1>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] cBioPortalData_2.4.5        MultiAssayExperiment_1.18.0
##  [3] SummarizedExperiment_1.22.0 Biobase_2.52.0             
##  [5] GenomicRanges_1.44.0        GenomeInfoDb_1.28.0        
##  [7] IRanges_2.26.0              S4Vectors_0.30.0           
##  [9] BiocGenerics_0.38.0         MatrixGenerics_1.4.0       
## [11] matrixStats_0.59.0          AnVIL_1.4.0                
## [13] dplyr_1.0.6                 BiocStyle_2.20.2           
## 
## loaded via a namespace (and not attached):
##   [1] rjson_0.2.20              ellipsis_0.3.2           
##   [3] rprojroot_2.0.2           futile.logger_1.4.3      
##   [5] XVector_0.32.0            fs_1.5.0                 
##   [7] rstudioapi_0.13           bit64_4.0.5              
##   [9] AnnotationDbi_1.54.1      fansi_0.5.0              
##  [11] xml2_1.3.2                codetools_0.2-18         
##  [13] splines_4.1.0             cachem_1.0.5             
##  [15] knitr_1.33                jsonlite_1.7.2           
##  [17] Rsamtools_2.8.0           dbplyr_2.1.1             
##  [19] png_0.1-7                 BiocManager_1.30.16      
##  [21] readr_1.4.0               compiler_4.1.0           
##  [23] httr_1.4.2                assertthat_0.2.1         
##  [25] Matrix_1.3-4              fastmap_1.1.0            
##  [27] limma_3.48.0              cli_2.5.0                
##  [29] formatR_1.11              htmltools_0.5.1.1        
##  [31] prettyunits_1.1.1         tools_4.1.0              
##  [33] glue_1.4.2                GenomeInfoDbData_1.2.6   
##  [35] rappdirs_0.3.3            Rcpp_1.0.6               
##  [37] rapiclient_0.1.3          pkgdown_1.6.1            
##  [39] vctrs_0.3.8               Biostrings_2.60.1        
##  [41] RJSONIO_1.3-1.4           RaggedExperiment_1.16.0  
##  [43] rtracklayer_1.52.0        xfun_0.24                
##  [45] stringr_1.4.0             ps_1.6.0                 
##  [47] RTCGAToolbox_2.22.1       rvest_1.0.0              
##  [49] lifecycle_1.0.0           restfulr_0.0.13          
##  [51] XML_3.99-0.6              zlibbioc_1.38.0          
##  [53] RCircos_1.2.1             ragg_1.1.3               
##  [55] hms_1.1.0                 lambda.r_1.2.4           
##  [57] yaml_2.2.1                curl_4.3.1               
##  [59] memoise_2.0.0             biomaRt_2.48.1           
##  [61] stringi_1.6.2             RSQLite_2.2.7            
##  [63] BiocIO_1.2.0              desc_1.3.0               
##  [65] GenomicDataCommons_1.16.0 GenomicFeatures_1.44.0   
##  [67] filelock_1.0.2            BiocParallel_1.26.0      
##  [69] rlang_0.4.11              pkgconfig_2.0.3          
##  [71] systemfonts_1.0.2         bitops_1.0-7             
##  [73] evaluate_0.14             lattice_0.20-44          
##  [75] purrr_0.3.4               GenomicAlignments_1.28.0 
##  [77] bit_4.0.4                 tidyselect_1.1.1         
##  [79] magrittr_2.0.1            bookdown_0.22            
##  [81] R6_2.5.0                  generics_0.1.0           
##  [83] DelayedArray_0.18.0       DBI_1.1.1                
##  [85] pillar_1.6.1              withr_2.4.2              
##  [87] survival_3.2-11           KEGGREST_1.32.0          
##  [89] RCurl_1.98-1.3            tibble_3.1.2             
##  [91] crayon_1.4.1              futile.options_1.0.1     
##  [93] utf8_1.2.1                BiocFileCache_2.0.0      
##  [95] rmarkdown_2.9             progress_1.2.2           
##  [97] grid_4.1.0                data.table_1.14.0        
##  [99] blob_1.2.1                digest_0.6.27            
## [101] tidyr_1.1.3               textshaping_0.3.5        
## [103] TCGAutils_1.12.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Levi Waldron, Marcel Ramos.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
