<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>cBioPortalData: User Start Guide • cBioPortalData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="cBioPortalData: User Start Guide">
<meta property="og:description" content="cBioPortalData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cBioPortalData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.4.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/cBioPortalData.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cBioPortalDataErrors.html">cBioPortalData: Data Build Errors</a>
    </li>
    <li>
      <a href="../articles/cBioPortalRClient.html">cBioPortalData: API Reference Guide for Devs</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/cBioPortalData/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="cBioPortalData_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>cBioPortalData: User Start Guide</h1>
                        <h4 class="author">Marcel Ramos &amp; Levi Waldron</h4>
            
            <h4 class="date">August 23, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/cBioPortalData/blob/master/vignettes/cBioPortalData.Rmd"><code>vignettes/cBioPortalData.Rmd</code></a></small>
      <div class="hidden name"><code>cBioPortalData.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">cBioPortalData</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">AnVIL</span><span class="op">)</span></code></pre></div>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The cBioPortal for Cancer Genomics <a href="https://cbioportal.org">website</a> is a great resource for interactive exploration of study datasets. However, it does not easily allow the analyst to obtain and further analyze the data.</p>
<p>We’ve developed the <code>cBioPortalData</code> package to fill this need to programmatically access the data resources available on the cBioPortal.</p>
<p>The <code>cBioPortalData</code> package provides an R interface for accessing the cBioPortal study data within the Bioconductor ecosystem.</p>
<p>It downloads study data from the cBioPortal API (the full API specification can be found here <a href="https://cbioportal.org/api" class="uri">https://cbioportal.org/api</a>) and uses Bioconductor infrastructure to cache and represent the data.</p>
<p>We use the <a href="https://dx.doi.org/10.1158/0008-5472.CAN-17-0344"><code>MultiAssayExperiment</code></a> (<span class="citation">@Ramos2017-er</span>) package to integrate, represent, and coordinate multiple experiments for the studies availble in the cBioPortal. This package in conjunction with <code>curatedTCGAData</code> give access to a large trove of publicly available bioinformatic data. Please see our publication <a href="https://dx.doi.org/10.1200/CCI.19.00119">here</a> (<span class="citation">@Ramos2020-ya</span>).</p>
<p>We demonstrate common use cases of <code>cBioPortalData</code> and <code>curatedTCGAData</code> during Bioconductor conference <a href="https://waldronlab.io/MultiAssayWorkshop/">workshops</a>.</p>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<div id="data-structures" class="section level3">
<h3 class="hasAnchor">
<a href="#data-structures" class="anchor"></a>Data Structures</h3>
<p>Data are provided as a single <code>MultiAssayExperiment</code> per study. The <code>MultiAssayExperiment</code> representation usually contains <code>SummarizedExperiment</code> objects for expression data and <code>RaggedExperiment</code> objects for mutation and CNV-type data. <code>RaggedExperiment</code> is a data class for representing ‘ragged’ genomic location data, meaning that the measurements per sample vary.</p>
<p>For more information, please see the <code>RaggedExperiment</code> and <code>SummarizedExperiment</code> vignettes.</p>
</div>
<div id="identifying-available-studies" class="section level3">
<h3 class="hasAnchor">
<a href="#identifying-available-studies" class="anchor"></a>Identifying available studies</h3>
<p>As we work through the data, there are some datasest that cannot be represented as <code>MultiAssayExperiment</code> objects. This can be due to a number of reasons such as the way the data is handled, presence of mis-matched identifiers, invalid data types, etc. To see what datasets are currently not building, we can look at the <code>studiesTable</code> dataset that comes with the package. We can invoke it using:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"studiesTable"</span>, package <span class="op">=</span> <span class="st">"cBioPortalData"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">studiesTable</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 6 columns
##       cancer_study_id             study_name            description
##           &lt;character&gt;            &lt;character&gt;            &lt;character&gt;
## 1       paac_jhu_2014 Acinar Cell Carcinom.. Whole exome sequenci..
## 2 mel_tsam_liang_2017 Acral Melanoma (TGEN.. Whole exome sequenci..
## 3     all_stjude_2015 Acute Lymphoblastic .. Comprehensive profil..
## 4     all_stjude_2016 Acute Lymphoblastic .. Whole-genome and/or ..
## 5       aml_ohsu_2018 Acute Myeloid Leukem.. Whole-exome sequenci..
## 6           laml_tcga Acute Myeloid Leukem.. TCGA Acute Myeloid L..
##                      URL pack_build api_build
##              &lt;character&gt;  &lt;logical&gt; &lt;logical&gt;
## 1                              TRUE      TRUE
## 2                             FALSE      TRUE
## 3                              TRUE      TRUE
## 4                              TRUE      TRUE
## 5                              TRUE      TRUE
## 6 http://gdac.broadins..       TRUE      TRUE</code></pre>
<p>The last two columns will show the availability of each <code>cancer_study_id</code> for either download method (<code>pack_build</code> for <code>cBioDataPack</code> and <code>api_build</code> for <code>cBioPortalData</code>).</p>
</div>
<div id="choosing-download-method" class="section level3">
<h3 class="hasAnchor">
<a href="#choosing-download-method" class="anchor"></a>Choosing download method</h3>
<p>There are two main user-facing functions for downloading data from the cBioPortal API.</p>
<ul>
<li><p><code>cBioDataPack</code> makes use of the tarball distribution of study data. This is useful when the user wants to download and analyze the entirety of the data as available from the cBioPortal.org website.</p></li>
<li><p><code>cBioPortalData</code> allows a more flexibile approach to obtaining study data based on the available parameters such as molecular profile identifiers. This option is useful for users who have a set of gene symbols or identifiers and would like to get a smaller subset of the data that correspond to a particular molecular profile.</p></li>
</ul>
</div>
</div>
<div id="two-main-functions" class="section level2">
<h2 class="hasAnchor">
<a href="#two-main-functions" class="anchor"></a>Two main functions</h2>
<div id="cbiodatapack-obtain-study-data-as-zipped-tarballs" class="section level3">
<h3 class="hasAnchor">
<a href="#cbiodatapack-obtain-study-data-as-zipped-tarballs" class="anchor"></a>cBioDataPack: Obtain Study Data as Zipped Tarballs</h3>
<p>This function will access the packaged data from  and return an integrative MultiAssayExperiment representation.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Use ask=FALSE for non-interactive use</span>
<span class="va">laml</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioDataPack.html">cBioDataPack</a></span><span class="op">(</span><span class="st">"laml_tcga"</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">laml</span></code></pre></div>
<pre><code>## A MultiAssayExperiment object of 13 listed
##  experiments with user-defined names and respective classes.
##  Containing an ExperimentList class object of length 13:
##  [1] cna_hg19.seg: RaggedExperiment with 13571 rows and 191 columns
##  [2] CNA: SummarizedExperiment with 24776 rows and 191 columns
##  [3] linear_CNA: SummarizedExperiment with 24776 rows and 191 columns
##  [4] methylation_hm27: SummarizedExperiment with 10968 rows and 194 columns
##  [5] methylation_hm450: SummarizedExperiment with 10968 rows and 194 columns
##  [6] mutations_extended: RaggedExperiment with 2584 rows and 197 columns
##  [7] mutations_mskcc: RaggedExperiment with 2584 rows and 197 columns
##  [8] RNA_Seq_expression_median: SummarizedExperiment with 19720 rows and 179 columns
##  [9] RNA_Seq_mRNA_median_all_sample_Zscores: SummarizedExperiment with 19720 rows and 179 columns
##  [10] RNA_Seq_mRNA_median_Zscores: SummarizedExperiment with 19719 rows and 179 columns
##  [11] RNA_Seq_v2_expression_median: SummarizedExperiment with 20531 rows and 173 columns
##  [12] RNA_Seq_v2_mRNA_median_all_sample_Zscores: SummarizedExperiment with 20531 rows and 173 columns
##  [13] RNA_Seq_v2_mRNA_median_Zscores: SummarizedExperiment with 20440 rows and 173 columns
## Functionality:
##  experiments() - obtain the ExperimentList instance
##  colData() - the primary/phenotype DataFrame
##  sampleMap() - the sample coordination DataFrame
##  `$`, `[`, `[[` - extract colData columns, subset, or experiment
##  *Format() - convert into a long or wide DataFrame
##  assays() - convert ExperimentList to a SimpleList of matrices
##  exportClass() - save all data to files</code></pre>
</div>
<div id="cbioportaldata-obtain-data-from-the-cbioportal-api" class="section level3">
<h3 class="hasAnchor">
<a href="#cbioportaldata-obtain-data-from-the-cbioportal-api" class="anchor"></a>cBioPortalData: Obtain data from the cBioPortal API</h3>
<p>This function provides a more flexible and granular way to request a MultiAssayExperiment object from a study ID, molecular profile, gene panel, sample list.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cbio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">cBioPortal</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">acc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortalData.html">cBioPortalData</a></span><span class="op">(</span>api <span class="op">=</span> <span class="va">cbio</span>, by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>, studyId <span class="op">=</span> <span class="st">"acc_tcga"</span>,
    genePanelId <span class="op">=</span> <span class="st">"IMPACT341"</span>,
    molecularProfileIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"acc_tcga_rppa"</span>, <span class="st">"acc_tcga_linear_CNA"</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<pre><code>## harmonizing input:
##   removing 1 colData rownames not in sampleMap 'primary'</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">acc</span></code></pre></div>
<pre><code>## A MultiAssayExperiment object of 2 listed
##  experiments with user-defined names and respective classes.
##  Containing an ExperimentList class object of length 2:
##  [1] acc_tcga_rppa: SummarizedExperiment with 57 rows and 46 columns
##  [2] acc_tcga_linear_CNA: SummarizedExperiment with 339 rows and 90 columns
## Functionality:
##  experiments() - obtain the ExperimentList instance
##  colData() - the primary/phenotype DataFrame
##  sampleMap() - the sample coordination DataFrame
##  `$`, `[`, `[[` - extract colData columns, subset, or experiment
##  *Format() - convert into a long or wide DataFrame
##  assays() - convert ExperimentList to a SimpleList of matrices
##  exportClass() - save all data to files</code></pre>
<p>Note. To avoid overloading the API service, the API was designed to only query a part of the study data. Therefore, the user is required to enter either a set of genes of interest or a gene panel identifier.</p>
</div>
</div>
<div id="clearing-the-cache" class="section level2">
<h2 class="hasAnchor">
<a href="#clearing-the-cache" class="anchor"></a>Clearing the cache</h2>
<div id="cbiodatapack" class="section level3">
<h3 class="hasAnchor">
<a href="#cbiodatapack" class="anchor"></a>cBioDataPack</h3>
<p>In cases where a download is interrupted, the user may experience a corrupt cache. The user can clear the cache for a particular study by using the <code>removeCache</code> function. Note that this function only works for data downloaded through the <code>cBioDataPack</code> function.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">removeCache</span><span class="op">(</span><span class="st">"laml_tcga"</span><span class="op">)</span></code></pre></div>
</div>
<div id="cbioportaldata" class="section level3">
<h3 class="hasAnchor">
<a href="#cbioportaldata" class="anchor"></a>cBioPortalData</h3>
<p>For users who wish to clear the entire <code>cBioPortalData</code> cache, it is recommended that they use:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/unlink.html">unlink</a></span><span class="op">(</span><span class="st">"~/.cache/cBioPortalData/"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="example-analysis-kaplan-meier-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#example-analysis-kaplan-meier-plot" class="anchor"></a>Example Analysis: Kaplan-Meier Plot</h2>
<p>We can use information in the <code>colData</code> to draw a K-M plot with a few variables from the <code>colData</code> slot of the <code>MultiAssayExperiment</code>. First, we load the necessary packages:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/therneau/survival">survival</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/survminer/index.html">survminer</a></span><span class="op">)</span></code></pre></div>
<p>We can check the data to lookout for any issues.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">laml</span><span class="op">)</span><span class="op">$</span><span class="va">OS_STATUS</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   0:LIVING 1:DECEASED 
##         67        133</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">laml</span><span class="op">)</span><span class="op">$</span><span class="va">OS_MONTHS</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "character"</code></pre>
<p>Now, we clean the data a bit to ensure that our variables are of the right type for the subsequent survival model fit.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">collaml</span> <span class="op">&lt;-</span> <span class="fu">colData</span><span class="op">(</span><span class="va">laml</span><span class="op">)</span>
<span class="va">collaml</span><span class="op">[</span><span class="va">collaml</span><span class="op">$</span><span class="va">OS_MONTHS</span> <span class="op">==</span> <span class="st">"[Not Available]"</span>, <span class="st">"OS_MONTHS"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span>
<span class="va">collaml</span><span class="op">$</span><span class="va">OS_MONTHS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">collaml</span><span class="op">$</span><span class="va">OS_MONTHS</span><span class="op">)</span>
<span class="fu">colData</span><span class="op">(</span><span class="va">laml</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">collaml</span></code></pre></div>
<p>We specify a simple survival model using <code>SEX</code> as a covariate and we draw the K-M plot.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/survival/man/survfit.html">survfit</a></span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/pkg/survival/man/Surv.html">Surv</a></span><span class="op">(</span><span class="va">OS_MONTHS</span>, <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">OS_STATUS</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">~</span> <span class="va">SEX</span>,
    data <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">laml</span><span class="op">)</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/survminer/man/ggsurvplot.html">ggsurvplot</a></span><span class="op">(</span><span class="va">fit</span>, data <span class="op">=</span> <span class="fu">colData</span><span class="op">(</span><span class="va">laml</span><span class="op">)</span>, risk.table <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="cBioPortalData_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<div id="data-update-requests" class="section level3">
<h3 class="hasAnchor">
<a href="#data-update-requests" class="anchor"></a>Data update requests</h3>
<p>If you are interested in a particular study dataset that is not currently building, please open an issue at our GitHub repository <a href="https://github.com/waldronlab/cBioPortalData/issues">location</a> and we will do our best to resolve the issues with either the data or the code.</p>
<p>We appreciate your feedback!</p>
</div>
</div>
</div>
<div id="sessioninfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>sessionInfo</h1>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] survminer_0.4.9             ggpubr_0.4.0               
##  [3] ggplot2_3.3.5               survival_3.2-12            
##  [5] cBioPortalData_2.4.8        MultiAssayExperiment_1.18.0
##  [7] SummarizedExperiment_1.22.0 Biobase_2.52.0             
##  [9] GenomicRanges_1.44.0        GenomeInfoDb_1.28.1        
## [11] IRanges_2.26.0              S4Vectors_0.30.0           
## [13] BiocGenerics_0.38.0         MatrixGenerics_1.4.2       
## [15] matrixStats_0.60.0          AnVIL_1.4.1                
## [17] dplyr_1.0.7                 BiocStyle_2.20.2           
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.3.1              backports_1.2.1          
##   [3] BiocFileCache_2.0.0       systemfonts_1.0.2        
##   [5] RCircos_1.2.1             splines_4.1.0            
##   [7] BiocParallel_1.26.1       TCGAutils_1.12.0         
##   [9] digest_0.6.27             htmltools_0.5.1.1        
##  [11] fansi_0.5.0               magrittr_2.0.1           
##  [13] memoise_2.0.0             tzdb_0.1.2               
##  [15] openxlsx_4.2.4            limma_3.48.3             
##  [17] Biostrings_2.60.2         readr_2.0.1              
##  [19] vroom_1.5.4               pkgdown_1.6.1            
##  [21] prettyunits_1.1.1         colorspace_2.0-2         
##  [23] blob_1.2.2                rvest_1.0.1              
##  [25] rappdirs_0.3.3            textshaping_0.3.5        
##  [27] haven_2.4.3               xfun_0.25                
##  [29] crayon_1.4.1              RCurl_1.98-1.4           
##  [31] jsonlite_1.7.2            RaggedExperiment_1.16.0  
##  [33] zoo_1.8-9                 glue_1.4.2               
##  [35] GenomicDataCommons_1.16.0 gtable_0.3.0             
##  [37] zlibbioc_1.38.0           XVector_0.32.0           
##  [39] DelayedArray_0.18.0       car_3.0-11               
##  [41] abind_1.4-5               scales_1.1.1             
##  [43] futile.options_1.0.1      DBI_1.1.1                
##  [45] rstatix_0.7.0             Rcpp_1.0.7               
##  [47] gridtext_0.1.4            xtable_1.8-4             
##  [49] progress_1.2.2            foreign_0.8-81           
##  [51] bit_4.0.4                 km.ci_0.5-2              
##  [53] httr_1.4.2                ellipsis_0.3.2           
##  [55] farver_2.1.0              pkgconfig_2.0.3          
##  [57] XML_3.99-0.7              rapiclient_0.1.3         
##  [59] dbplyr_2.1.1              utf8_1.2.2               
##  [61] RJSONIO_1.3-1.5           labeling_0.4.2           
##  [63] tidyselect_1.1.1          rlang_0.4.11             
##  [65] AnnotationDbi_1.54.1      munsell_0.5.0            
##  [67] cellranger_1.1.0          tools_4.1.0              
##  [69] cachem_1.0.6              cli_3.0.1                
##  [71] generics_0.1.0            RSQLite_2.2.8            
##  [73] broom_0.7.9               evaluate_0.14            
##  [75] stringr_1.4.0             fastmap_1.1.0            
##  [77] yaml_2.2.1                ragg_1.1.3               
##  [79] knitr_1.33                bit64_4.0.5              
##  [81] fs_1.5.0                  zip_2.2.0                
##  [83] survMisc_0.5.5            purrr_0.3.4              
##  [85] KEGGREST_1.32.0           formatR_1.11             
##  [87] xml2_1.3.2                biomaRt_2.48.3           
##  [89] compiler_4.1.0            filelock_1.0.2           
##  [91] curl_4.3.2                png_0.1-7                
##  [93] ggsignif_0.6.2            tibble_3.1.3             
##  [95] stringi_1.7.3             highr_0.9                
##  [97] futile.logger_1.4.3       GenomicFeatures_1.44.1   
##  [99] desc_1.3.0                forcats_0.5.1            
## [101] lattice_0.20-44           Matrix_1.3-4             
## [103] markdown_1.1              KMsurv_0.1-5             
## [105] RTCGAToolbox_2.22.1       vctrs_0.3.8              
## [107] pillar_1.6.2              lifecycle_1.0.0          
## [109] BiocManager_1.30.16       data.table_1.14.0        
## [111] bitops_1.0-7              rtracklayer_1.52.1       
## [113] R6_2.5.1                  BiocIO_1.2.0             
## [115] bookdown_0.23             gridExtra_2.3            
## [117] rio_0.5.27                codetools_0.2-18         
## [119] lambda.r_1.2.4            assertthat_0.2.1         
## [121] rprojroot_2.0.2           rjson_0.2.20             
## [123] withr_2.4.2               GenomicAlignments_1.28.0 
## [125] Rsamtools_2.8.0           GenomeInfoDbData_1.2.6   
## [127] ggtext_0.1.1              hms_1.1.0                
## [129] grid_4.1.0                tidyr_1.1.3              
## [131] rmarkdown_2.10            carData_3.0-4            
## [133] restfulr_0.0.13</code></pre>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Levi Waldron, Marcel Ramos.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
