<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>cBioPortalData: User Start Guide â€¢ cBioPortalData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="cBioPortalData: User Start Guide">
<meta property="og:description" content="cBioPortalData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cBioPortalData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.2.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/cBioPortalData.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cBioPortalRClient.html">cBioPortalData: API Reference Guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/cBioPortalData/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="cBioPortalData_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>cBioPortalData: User Start Guide</h1>
                        <h4 class="author">Marcel Ramos</h4>
            
            <h4 class="date">February 03, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/cBioPortalData/blob/master/vignettes/cBioPortalData.Rmd"><code>vignettes/cBioPortalData.Rmd</code></a></small>
      <div class="hidden name"><code>cBioPortalData.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">cBioPortalData</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">AnVIL</span><span class="op">)</span></code></pre></div>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This vignette lays out the two main user-facing functions for downloading and representing data from the cBioPortal API. <code>cBioDataPack</code> makes use of the legacy distribution data method in <code>cBioPortal</code> (via tarballs). <code>cBioPortalData</code> allows for a more flexibile approach to obtaining data based on several available parameters including available molecular profiles.</p>
<div id="two-main-interfaces" class="section level2">
<h2 class="hasAnchor">
<a href="#two-main-interfaces" class="anchor"></a>Two main interfaces</h2>
<div id="cbiodatapack-obtain-study-data-as-zipped-tarballs" class="section level3">
<h3 class="hasAnchor">
<a href="#cbiodatapack-obtain-study-data-as-zipped-tarballs" class="anchor"></a>cBioDataPack: Obtain Study Data as Zipped Tarballs</h3>
<p>This function will access the packaged data from  and return an integrative MultiAssayExperiment representation.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Use ask=FALSE for non-interactive use</span>
<span class="fu"><a href="../reference/cBioDataPack.html">cBioDataPack</a></span><span class="op">(</span><span class="st">"laml_tcga"</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code>## A MultiAssayExperiment object of 13 listed
##  experiments with user-defined names and respective classes.
##  Containing an ExperimentList class object of length 13:
##  [1] cna_hg19.seg: RaggedExperiment with 13571 rows and 191 columns
##  [2] CNA: SummarizedExperiment with 24776 rows and 191 columns
##  [3] linear_CNA: SummarizedExperiment with 24776 rows and 191 columns
##  [4] methylation_hm27: SummarizedExperiment with 10919 rows and 194 columns
##  [5] methylation_hm450: SummarizedExperiment with 10919 rows and 194 columns
##  [6] mutations_extended: RaggedExperiment with 2584 rows and 197 columns
##  [7] mutations_mskcc: RaggedExperiment with 2584 rows and 197 columns
##  [8] RNA_Seq_expression_median: SummarizedExperiment with 19720 rows and 179 columns
##  [9] RNA_Seq_mRNA_median_all_sample_Zscores: SummarizedExperiment with 19720 rows and 179 columns
##  [10] RNA_Seq_mRNA_median_Zscores: SummarizedExperiment with 19719 rows and 179 columns
##  [11] RNA_Seq_v2_expression_median: SummarizedExperiment with 20531 rows and 173 columns
##  [12] RNA_Seq_v2_mRNA_median_all_sample_Zscores: SummarizedExperiment with 20531 rows and 173 columns
##  [13] RNA_Seq_v2_mRNA_median_Zscores: SummarizedExperiment with 20440 rows and 173 columns
## Functionality:
##  experiments() - obtain the ExperimentList instance
##  colData() - the primary/phenotype DataFrame
##  sampleMap() - the sample coordination DataFrame
##  `$`, `[`, `[[` - extract colData columns, subset, or experiment
##  *Format() - convert into a long or wide DataFrame
##  assays() - convert ExperimentList to a SimpleList of matrices
##  exportClass() - save all data to files</code></pre>
</div>
<div id="cbioportaldata-obtain-data-from-the-cbioportal-api" class="section level3">
<h3 class="hasAnchor">
<a href="#cbioportaldata-obtain-data-from-the-cbioportal-api" class="anchor"></a>cBioPortalData: Obtain data from the cBioPortal API</h3>
<p>This function provides a more flexible and granular way to request a MultiAssayExperiment object from a study ID, molecular profile, gene panel, sample list.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cbio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortal.html">cBioPortal</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">acc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cBioPortalData.html">cBioPortalData</a></span><span class="op">(</span>api <span class="op">=</span> <span class="va">cbio</span>, by <span class="op">=</span> <span class="st">"hugoGeneSymbol"</span>, studyId <span class="op">=</span> <span class="st">"acc_tcga"</span>,
    genePanelId <span class="op">=</span> <span class="st">"IMPACT341"</span>,
    molecularProfileIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"acc_tcga_rppa"</span>, <span class="st">"acc_tcga_linear_CNA"</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<pre><code>## harmonizing input:
##   removing 1 colData rownames not in sampleMap 'primary'</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">acc</span></code></pre></div>
<pre><code>## A MultiAssayExperiment object of 2 listed
##  experiments with user-defined names and respective classes.
##  Containing an ExperimentList class object of length 2:
##  [1] acc_tcga_rppa: SummarizedExperiment with 57 rows and 46 columns
##  [2] acc_tcga_linear_CNA: SummarizedExperiment with 339 rows and 90 columns
## Functionality:
##  experiments() - obtain the ExperimentList instance
##  colData() - the primary/phenotype DataFrame
##  sampleMap() - the sample coordination DataFrame
##  `$`, `[`, `[[` - extract colData columns, subset, or experiment
##  *Format() - convert into a long or wide DataFrame
##  assays() - convert ExperimentList to a SimpleList of matrices
##  exportClass() - save all data to files</code></pre>
</div>
</div>
<div id="clearing-the-cache" class="section level2">
<h2 class="hasAnchor">
<a href="#clearing-the-cache" class="anchor"></a>Clearing the cache</h2>
<div id="cbiodatapack" class="section level3">
<h3 class="hasAnchor">
<a href="#cbiodatapack" class="anchor"></a>cBioDataPack</h3>
<p>In cases where a download is interrupted, the user may experience a corrupt cache. The user can clear the cache for a particular study by using the <code>removeCache</code> function. Note that this function only works for data downloaded through the <code>cBioDataPack</code> function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">removeCache</span><span class="op">(</span><span class="st">"laml_tcga"</span><span class="op">)</span></code></pre></div>
</div>
<div id="cbioportaldata" class="section level3">
<h3 class="hasAnchor">
<a href="#cbioportaldata" class="anchor"></a>cBioPortalData</h3>
<p>For users who wish to clear the entire <code>cBioPortalData</code> cache, it is recommended that they use:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/unlink.html">unlink</a></span><span class="op">(</span><span class="st">"~/.cache/cBioPortalData/"</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
<div id="sessioninfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>sessionInfo</h1>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.1 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] cBioPortalData_2.2.7        MultiAssayExperiment_1.16.0
##  [3] SummarizedExperiment_1.20.0 Biobase_2.50.0             
##  [5] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
##  [7] IRanges_2.24.1              S4Vectors_0.28.1           
##  [9] BiocGenerics_0.36.0         MatrixGenerics_1.2.1       
## [11] matrixStats_0.58.0          AnVIL_1.2.0                
## [13] dplyr_1.0.4                 BiocStyle_2.18.1           
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-6              fs_1.5.0                 
##  [3] bit64_4.0.5               progress_1.2.2           
##  [5] httr_1.4.2                rprojroot_2.0.2          
##  [7] GenomicDataCommons_1.14.0 tools_4.0.3              
##  [9] R6_2.5.0                  DBI_1.1.1                
## [11] withr_2.4.1               prettyunits_1.1.1        
## [13] tidyselect_1.1.0          TCGAutils_1.10.0         
## [15] bit_4.0.4                 curl_4.3                 
## [17] compiler_4.0.3            cli_2.3.0                
## [19] rvest_0.3.6               textshaping_0.2.1        
## [21] formatR_1.7               xml2_1.3.2               
## [23] desc_1.2.0                DelayedArray_0.16.1      
## [25] rtracklayer_1.50.0        bookdown_0.21            
## [27] readr_1.4.0               askpass_1.1              
## [29] rappdirs_0.3.3            pkgdown_1.6.1            
## [31] rapiclient_0.1.3          RCircos_1.2.1            
## [33] Rsamtools_2.6.0           systemfonts_1.0.0        
## [35] stringr_1.4.0             digest_0.6.27            
## [37] rmarkdown_2.6             XVector_0.30.0           
## [39] pkgconfig_2.0.3           htmltools_0.5.1.1        
## [41] dbplyr_2.0.0              fastmap_1.1.0            
## [43] limma_3.46.0              rlang_0.4.10             
## [45] rstudioapi_0.13           RSQLite_2.2.3            
## [47] generics_0.1.0            jsonlite_1.7.2           
## [49] BiocParallel_1.24.1       RCurl_1.98-1.2           
## [51] magrittr_2.0.1            GenomeInfoDbData_1.2.4   
## [53] futile.logger_1.4.3       Matrix_1.3-2             
## [55] Rcpp_1.0.6                lifecycle_0.2.0          
## [57] stringi_1.5.3             yaml_2.2.1               
## [59] RaggedExperiment_1.14.1   RJSONIO_1.3-1.4          
## [61] zlibbioc_1.36.0           BiocFileCache_1.14.0     
## [63] grid_4.0.3                blob_1.2.1               
## [65] crayon_1.4.0              lattice_0.20-41          
## [67] Biostrings_2.58.0         splines_4.0.3            
## [69] GenomicFeatures_1.42.1    hms_1.0.0                
## [71] ps_1.5.0                  knitr_1.31               
## [73] pillar_1.4.7              codetools_0.2-18         
## [75] biomaRt_2.46.2            futile.options_1.0.1     
## [77] XML_3.99-0.5              glue_1.4.2               
## [79] evaluate_0.14             lambda.r_1.2.4           
## [81] data.table_1.13.6         BiocManager_1.30.10      
## [83] vctrs_0.3.6               tidyr_1.1.2              
## [85] openssl_1.4.3             purrr_0.3.4              
## [87] assertthat_0.2.1          cachem_1.0.1             
## [89] xfun_0.20                 ragg_0.4.1               
## [91] survival_3.2-7            tibble_3.0.6             
## [93] RTCGAToolbox_2.20.0       GenomicAlignments_1.26.0 
## [95] AnnotationDbi_1.52.0      memoise_2.0.0            
## [97] ellipsis_0.3.1</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Levi Waldron, Marcel Ramos.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
